x/batching/keeper/data_result.go

```diff
@@ -12,31 +12,31 @@ import (
 // SetDataResultForBatching stores a data result so that it is ready
 // to be batched.
 func (k Keeper) SetDataResultForBatching(ctx context.Context, result types.DataResult) error {
-	return k.dataResults.Set(ctx, collections.Join(false, result.DrId), result)
+	return k.dataResults.Set(ctx, collections.Join3(false, result.DrId, result.DrBlockHeight), result)
 }
 
 // MarkDataResultAsBatched removes the "unbatched" variant of the given
 // data result and stores a "batched" variant.
 func (k Keeper) MarkDataResultAsBatched(ctx context.Context, result types.DataResult, batchNum uint64) error {
-	err := k.SetBatchAssignment(ctx, result.DrId, batchNum)
+	err := k.SetBatchAssignment(ctx, result.DrId, result.DrBlockHeight, batchNum)
 	if err != nil {
 		return err
 	}
-	err = k.dataResults.Remove(ctx, collections.Join(false, result.DrId))
+	err = k.dataResults.Remove(ctx, collections.Join3(false, result.DrId, result.DrBlockHeight))
 	if err != nil {
 		return err
 	}
-	return k.dataResults.Set(ctx, collections.Join(true, result.DrId), result)
+	return k.dataResults.Set(ctx, collections.Join3(true, result.DrId, result.DrBlockHeight), result)
 }
 
 // GetDataResult returns a data result given the associated data request's
-// ID.
-func (k Keeper) GetDataResult(ctx context.Context, dataReqID string) (*types.DataResult, error) {
-	dataResult, err := k.dataResults.Get(ctx, collections.Join(false, dataReqID))
+// ID and height.
+func (k Keeper) GetDataResult(ctx context.Context, dataReqID string, dataReqHeight uint64) (*types.DataResult, error) {
+	dataResult, err := k.dataResults.Get(ctx, collections.Join3(false, dataReqID, dataReqHeight))
 	if err != nil {
 		if errors.Is(err, collections.ErrNotFound) {
 			// Look among batched data requests.
-			dataResult, err := k.dataResults.Get(ctx, collections.Join(true, dataReqID))
+			dataResult, err := k.dataResults.Get(ctx, collections.Join3(true, dataReqID, dataReqHeight))
 			if err != nil {
 				if errors.Is(err, collections.ErrNotFound) {
 					return nil, nil
```

Comment:
Same issue here. Perhaps we shouldn't catch the `ErrNotFound` error? I'm also worried about inconsistencies in dealing with this error since most of our getters don't catch this error